lib/galaxy/webapps/galaxy/controllers/user.py

```diff
@@ -82,33 +82,35 @@ def __autoregistration(self, trans, login, password):
         """
         Does the autoregistration if enabled. Returns a message
         """
-        autoreg = trans.app.auth_manager.check_auto_registration(trans, login, password)
+        try :
+            autoreg = trans.app.auth_manager.check_auto_registration(trans, login, password)
+        except Conflict as conflict:
+            message = "Auto-registration failed, {}".format(conflict)
+            autoreg = None
         user = None
-        if autoreg["auto_reg"]:
-            email = autoreg["email"]
-            username = autoreg["username"]
-            message = " ".join([validate_email(trans, email, allow_empty=True),
-                                validate_publicname(trans, username)]).rstrip()
-            if not message:
-                user = self.user_manager.create(email=email, username=username, password="")
-                if trans.app.config.user_activation_on:
-                    self.user_manager.send_activation_email(trans, email, username)
-                # The handle_user_login() method has a call to the history_set_default_permissions() method
-                # (needed when logging in with a history), user needs to have default permissions set before logging in
-                if not trans.user_is_admin:
-                    trans.handle_user_login(user)
-                    trans.log_event("User (auto) created a new account")
-                    trans.log_event("User logged in")
-                if "attributes" in autoreg and "roles" in autoreg["attributes"]:
-                    self.__handle_role_and_group_auto_creation(
-                        trans, user, autoreg["attributes"]["roles"],
-                        auto_create_groups=autoreg["auto_create_groups"],
-                        auto_create_roles=autoreg["auto_create_roles"],
+        if autoreg :
+            if autoreg["auto_reg"]:
```

Comment:
You can combine the 2 ifs and save one indentation level:
```python
        if autoreg and autoreg["auto_reg"]:
```

Reply:
I thought about it . However, its imply that the next else have to be updated to an  `eleif autoreg: ` ( to be able to discriminate the case when autoreg exist and autoreg["auto_reg"] not).  That make this piece of code less intuitive because at the first reading,  its look like the 'same' check is done twice.
That being said, I can go for the 'elif cascade' if you think that indentation level save is wroth the change.

Reply:
I don't see an `else` at level of `if autoreg:` (and I don't think one is needed!), so I'm not sure I understand your comment.

Reply:
My comment concern the `else: message = "No such user or invalid password."` If I go for a ` if autoreg and autoreg["auto_reg"]:` like you suggest this one cannot stay in place as such.

Reply:
I see, thanks for explaining. I'd suggest an early return then, see suggestion above.